sklearn/_loss/loss.py

```diff
@@ -968,6 +1080,42 @@ def predict_proba(self, raw_prediction):
         proba[:, 0] = 1 - proba[:, 1]
         return proba
 
+    def loss_array_api(self, y_true, raw_prediction, sample_weight=None, xp=None):
+        xp, _ = get_namespace(y_true, raw_prediction, sample_weight, xp=xp)
+        loss = xp.log1p(xp.exp(raw_prediction)) - y_true * raw_prediction
+        if sample_weight is not None:
+            loss *= sample_weight
+        return loss
+
+    def gradient_array_api(self, y_true, raw_prediction, sample_weight=None, xp=None):
+        xp, _ = get_namespace(y_true, raw_prediction, sample_weight, xp=xp)
+        grad = xp.where(
+            raw_prediction > -37,
+            ((1 - y_true) - y_true * xp.exp(-raw_prediction))
+            / (1 + xp.exp(-raw_prediction)),
```

Comment:
Let's use a local variable to store the result of `xp.exp(-raw_prediction)` instead of computing it twice.